server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1298,7 +1298,12 @@ private static List<String> staleRootBlobs(RepositoryData repositoryData, Set<St
                 return allSnapshotIds.contains(foundUUID) == false;
             } else if (blob.startsWith(INDEX_FILE_PREFIX)) {
                 // TODO: Include the current generation here once we remove keeping index-(N-1) around from #writeIndexGen
-                return repositoryData.getGenId() > Long.parseLong(blob.substring(INDEX_FILE_PREFIX.length()));
+                try {
+                    return repositoryData.getGenId() > Long.parseLong(blob.substring(INDEX_FILE_PREFIX.length()));
+                } catch (NumberFormatException nfe) {
+                    // odd case of an extra file with the index- prefix that we can't identify
+                    return false;
```

Comment:
Should we log that we discovered unexpected file in repository?

Reply:
I'm mildly worried this could get really spammy in some cases and we do already log this specific case in `org.elasticsearch.repositories.blobstore.BlobStoreRepository#latestGeneration` actually.
This is hopefully an edge case to the point where we won't have to worry much about it :)